### PR TITLE
Prevent inappropriate rodsadmin downgrades in same zone (main) (updated)

### PR DESCRIPTION
**Pull request by korydraughn**
Replaces PR #6466

Will run tests soon.

### CODE REVIEW

File: scripts/irods/test/test_iadmin.py [hunk 1]
```diff
@@ -2180,38 +2176,42 @@ def test_moduser_type_rodsadmin_rodsuser(self):
...
         self.assertEqual('rodsuser', lib.get_user_type(self.admin, self.username))
         self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'type', 'invalid_user_type'],
                                    'STDERR_SINGLELINE', 'CAT_INVALID_USER_TYPE')
         self.assertEqual('rodsuser', lib.get_user_type(self.admin, self.username))
 
+    def test_downgrade_of_service_account_user_is_not_allowed__issue_6127(self):
+        """Test downgrading of service account user's type from rodsadmin to other supported types."""
+        # rodsadmin -> rodsuser
+        self.assertEqual('rodsadmin', lib.get_user_type(self.admin, 'rods'))
+        downgrade_error_msg = 'Cannot downgrade another rodsadmin'
```
Comment (alanking) [inline scripts/irods/test/test_iadmin.py:2201]: Should we include more words or a comment here? I got a little confused when I saw this expected message next to the title of the test. It makes it look like no `rodsadmin` is allowed to downgrade any other `rodsadmin`. Maybe that's just me :)

File: scripts/irods/test/test_iadmin.py [hunk 2]
```diff
@@ -2164,10 +2164,6 @@ def test_moduser_type_rodsadmin_rodsuser(self):
         self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'type', 'rodsadmin'])
         self.assertEqual('rodsadmin', lib.get_user_type(self.admin, self.username))
 
-        # rodsadmin -> rodsuser
-        self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'type', 'rodsuser'])
-        self.assertEqual('rodsuser', lib.get_user_type(self.admin, self.username))
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply scripts/irods/test/test_iadmin.py:2169]: Nice catch. Will restore.

File: scripts/irods/test/test_iadmin.py (hunk 1 above)
Comment (korydraughn) [reply scripts/irods/test/test_iadmin.py:2201]: Will use more words.

File: scripts/irods/test/test_iadmin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (korydraughn) [reply scripts/irods/test/test_iadmin.py]: Done

File: scripts/irods/test/test_iadmin.py (hunk 1 above)
Comment (korydraughn) [reply scripts/irods/test/test_iadmin.py]: Done

**Comment by korydraughn** [comment]
All core/unit tests passed.

All that is left is to clean up and add the #'s.

**Review by alanking — DISMISSED** [review]
\# it up

**Comment by korydraughn** [comment]
Added the #'s

**Review by alanking — APPROVED** [review]

**Pull request merged**